euroc_data_publisher:
  ros__parameters:
    # Camera 1 Configuration
    cam1:
      sensor_type: "camera"
      comment: "VI-Sensor cam1 (MT9M034)"
      
      # Publishing rate
      rate_hz: 20.0
      
      # Camera resolution [width, height]
      resolution: [752, 480]
      
      # Camera model and intrinsics
      camera_model: "pinhole"
      intrinsics: [457.587, 456.134, 379.999, 255.238]  # [fu, fv, cu, cv]
      
      # Distortion parameters
      distortion_model: "radial-tangential"
      distortion_coefficients: [-0.28368365, 0.07451284, -0.00010473, -3.55590700e-05]
      
      # Sensor extrinsics wrt. the body-frame (row-major 4x4 matrix)
      T_BS: [
        0.0125552670891, -0.999755099723, 0.0182237714554, -0.0198435579556,
        0.999598781151, 0.0130119051815, 0.0251588363115, 0.0453689425024,
        -0.0253898008918, 0.0179005838253, 0.999517347078, 0.00786212447038,
        0.0, 0.0, 0.0, 1.0
      ]
